psi4/src/psi4/libscf_solver/hf.h

```diff
@@ -146,7 +148,7 @@ class HF : public Wavefunction {
     /// DIIS manager intiialized?
     bool initialized_diis_manager_;
     /// DIIS manager for all SCF wavefunctions
-    std::shared_ptr<DIISManager> diis_manager_;
+    py::object diis_manager_;
```

Comment:
Why not use the new DIISManager class?

Reply:
Because then the Python-side needs to manipulate a C-side class which is really a wrapper to another Python-class.

I don't think there's a good solution to this: using the Python object makes C-syntax clunky, and using the C-wrapper makes Py-calls go through C++ to access Python. I don't have a firm stance on which is better.

Reply:
Ah, ok, makes sense.